lib/trie/trie.go

```diff
@@ -167,675 +177,895 @@ func (t *Trie) Entries() map[string][]byte {
 	return entries(t.root, nil, make(map[string][]byte))
 }
 
-func entries(current Node, prefix []byte, kv map[string][]byte) map[string][]byte {
-	switch c := current.(type) {
-	case *node.Branch:
-		if c.Value != nil {
-			kv[string(codec.NibblesToKeyLE(append(prefix, c.Key...)))] = c.Value
-		}
-		for i, child := range c.Children {
-			entries(child, append(prefix, append(c.Key, byte(i))...), kv)
-		}
-	case *node.Leaf:
-		kv[string(codec.NibblesToKeyLE(append(prefix, c.Key...)))] = c.Value
+func entries(parent Node, prefix []byte, kv map[string][]byte) map[string][]byte {
+	if parent == nil {
+		return kv
+	}
+
+	if parent.Type() == node.LeafType {
+		parentKey := parent.GetKey()
+		fullKeyNibbles := make([]byte, 0, len(prefix)+len(parentKey))
+		fullKeyNibbles = append(fullKeyNibbles, prefix...)
+		fullKeyNibbles = append(fullKeyNibbles, parentKey...)
```

Comment:
```suggestion
		fullKeyNibbles := append(prefix, parentKey...)
```

I'm curious what the difference between these two functionally is? the values aren't being copied either way, so it seems to me it's the same?

Reply:
Super micro optimization, see [this go-guild Slack comment](https://chainsafeio.slack.com/archives/C022YPNK4HF/p1642803472018200) or benchmark it yourself with https://go.dev/play/p/ZuD_AWdqZxo 😉 

Since I found out recently, I couldn't hold myself to add it here 😆

Reply:
Added a benchmark in https://github.com/ChainSafe/gossamer/pull/2255/commits/9757382edea4068a4463229b5f937fcf84e3b701 with a helper function to concatenate slices.